### PR TITLE
RIP-2 Storage Layout

### PR DESCRIPTION
**Pull request by FintanH**
Add RIP-2 describing the layout of the git storage for the Heartwood protocol.

### CODE REVIEW

File: 0002-storage-layout.md [hunk 1]
```diff
@@ -0,0 +1,518 @@
...
+Storage Home
+------------
+
+It is trivial to store `git` repositories, locally, on disk. It is
+also efficient to do so. The storage must be stored under the name
+`storage` found under one of two base directories.
+
+If the `RAD_HOME` environment variable is set, the directory it points
+to will be used as the base directory. If `RAD_HOME` is set it will
+take priority as the base directory.
```
Comment (keepsimple1) [inline 0002-storage-layout.md:77]: Is the last sentence repeating the previous one?

File: 0002-storage-layout.md [hunk 2]
```diff
@@ -0,0 +1,518 @@
...
+
+| Value                        | Example                       |
+| ---------------------------- | ------------------------------|
+| `$RAD_HOME/.radicle/storage` | /home/alice/MyRadicle/storage |
```
Comment (keepsimple1) [inline 0002-storage-layout.md:81]: What is the value of `$RAD_HOME` in the above? I don't understand how the full path ends up not in `.radicle`?

File: 0002-storage-layout.md [hunk 3]
```diff
@@ -0,0 +1,518 @@
...
+
+To have this separation, the [gitnamespaces][namespaces] feature of
+`git` is used. For each peer, including the local operator, their
+unique identifier is used as the namespace within the `<resource>`
+repository. For now, this unique identifier is assumed to be a
+hash-encoded value of the peer's public key, denoted `<pubkey>` from
+here onwards.
+
+This means that a peer's references will be scoped by
+`refs/namespaces/<pubkey>`. We demonstrate this organisation below:
```
Comment (keepsimple1) [inline 0002-storage-layout.md:127]: Does this rule apply to the local user / node?  I.e. do the `peers` include the local node as well?

File: 0002-storage-layout.md [hunk 4]
```diff
@@ -0,0 +1,518 @@
...
+Interoperability with `git`
+---------------------------
+
+There are two aspects for interoperability with `git`:
+1. Remote replication
+2. Linking to a working copy
+
+In the next sections we will cover how the above layout works.
+
+### Remote Replication
```
Comment (keepsimple1) [inline 0002-storage-layout.md:192]: nit:  it seems there are no `##` headers before this line?

File: 0002-storage-layout.md (hunk 1 above)
Comment (FintanH) [reply 0002-storage-layout.md:77]: Oh yes, thank you :bow:

File: 0002-storage-layout.md (hunk 2 above)
Comment (FintanH) [reply 0002-storage-layout.md:81]: It's an environment variable, so the user can set it to whatever they want. But actually, it shouldn't end up in `.radicle` :) It should in fact be like this:
```suggestion
| `$RAD_HOME/storage` | /home/alice/MyRadicle/storage |
```

File: 0002-storage-layout.md (hunk 3 above)
Comment (FintanH) [reply 0002-storage-layout.md:127]: I point this out above https://github.com/radicle-dev/rips/pull/1/files#diff-0ccf2baf8bebd800e0e6c8723c5dba6b925263f9e64b48643a619941fc35f4cdR120

File: 0002-storage-layout.md (hunk 4 above)
Comment (FintanH) [reply 0002-storage-layout.md:192]: I followed suit from the first RIP where it uses a series of `-` instead of `##`

File: 0002-storage-layout.md [hunk 5]
```diff
@@ -0,0 +1,517 @@
...
+
+RIP #2: Storage Layout
+======================
+
+Introduction
+------------
+
+One of the key components of the Radicle network is the storage
+layer. A key aspect of this component is that it is local-first. It
+must be able to hold the view of the local operator's view of a
```
Comment (cloudhead) [inline 0002-storage-layout.md:18]:
```suggestion
must be able to hold the local operator's view of a
```

File: 0002-storage-layout.md [hunk 6]
```diff
@@ -0,0 +1,517 @@
...
+
+The storage layer must also be designed for efficient replication of
+data between peers in the network. For this reason, `git` is used as
+the underlying database as it already efficiently exchanges data
+between two machines ([Transfer Protocols][git-protocol]).
+
+
+With the above in mind, this RIP proposes a storage layer that
+fulfills the following requirements:
+1. It can be stored, locally, on disk.
```
Comment (cloudhead) [inline 0002-storage-layout.md:39]: I'd rephrase this to carry more meaning, into something like "It is able to keep a local copy of the entire working dataset"

File: 0002-storage-layout.md [hunk 7]
```diff
@@ -0,0 +1,517 @@
...
+        * [Setup](#setup)
+        * [Pushing Changes](#pushing-changes)
+        * [Fetching Changes](#fetching-changes)
+        * [Different Peers](#different-peers)
+        * [Non-global Tags](#non-global-tags)
+* [Credits](#credits)
+* [Copyright](#copyright)
+
+Storage Home
+------------
```
Comment (cloudhead) [inline 0002-storage-layout.md:69]: Do you think it might be better to avoid specifying RAD_HOME entirely here? In which case we could say something like `let '/storage' be a directory under which all radicle git state shall be stored`...

Because it seems like a bit of a distraction when what we're interested in is more what is *in* the storage folder.

File: 0002-storage-layout.md [hunk 8]
```diff
@@ -0,0 +1,517 @@
...
+    …
+  <resource>/
+    …
+```
+
+Note that top-level references may still exist,
+i.e. `storage/<resource>/refs/{heads, tags}`. These namespaces must be
+reserved for canoncial references -- references that are agreed,
+collaboratively, as published and stable. How canonical references are
+decided and written is left for further RIP.
```
Comment (cloudhead) [inline 0002-storage-layout.md:154]: Actually there is one thing we could say here, it's that these references are *not* replicated. They are set locally, according to some heuristics that are TBD.

File: 0002-storage-layout.md [hunk 9]
```diff
@@ -0,0 +1,517 @@
...
+```
+
+Interoperability with `git`
+---------------------------
+
+There are two aspects for interoperability with `git`:
+1. Remote replication
+2. Linking to a working copy
+
+In the next sections we will cover how the above layout works.
```
Comment (cloudhead) [inline 0002-storage-layout.md:189]:
```suggestion
In the next sections we will cover how the above works with the storage layout works.
```
Makes more sense to me?

File: 0002-storage-layout.md [hunk 10]
```diff
@@ -0,0 +1,517 @@
...
+cryptographic verification of fetched data is out of scope for this
+RIP.
+
+### Working Copy
+
+A working copy is a local replica of the resource -- that is each
+working copy corresponds to one resource in the storage -- where the
+operator makes their own changes to the code base. This can be
+compared to the way one would `git clone` from a mirrored repository,
+e.g. GitHub, GitLab, etc. They can then make changes and push to the
```
Comment (cloudhead) [inline 0002-storage-layout.md:207]:
```suggestion
e.g. GitHub, GitLab, etc. One can then make changes and push to the
```

File: 0002-storage-layout.md [hunk 11]
```diff
@@ -0,0 +1,517 @@
...
+mirror. In the case of Radicle, fetching and pushing changes is
+between the working copy and the local-first storage.
+
+The connection between the working copy and the storage is maintained
+by a series of `git` [remotes][git-remotes]. Each `git` remote
+represents a single peer -- or namespace -- for that resource.
+
```
Comment (cloudhead) [inline 0002-storage-layout.md:214]: I'd add here something like:
> One remote in particular is always setup when a working copy is created: the `rad` remote, which allows operators to push to their own copy of the resource.

File: 0002-storage-layout.md [hunk 12]
```diff
@@ -0,0 +1,517 @@
...
+```
+
+To `git fetch` or `git push`, it is necessary to state the namespace
+that is being used for the operation. This can be achieved using `git
+--namespace=<ns>` or `GIT_NAMESPACE=<ns> git`. Unfortunately, this
+does not disallow the pushing to other peer's namespaces, nor allows
+signing pushed references to one's own namespace. This is discussed in
+[Future Work](#Future-Work).
+
+To ensure `gitnamespaces` worked as expected the author ran through a
```
Comment (cloudhead) [inline 0002-storage-layout.md:245]:
```suggestion
To ensure `gitnamespaces` work as expected the author ran through a
```

File: 0002-storage-layout.md [hunk 13]
```diff
@@ -0,0 +1,517 @@
...
+In [Working Copy](#Working-Copy), it was discussed that there is no
+way for a `git` remote to be configured so that it is aware of extra
+logic, such as which `refs/namespaces` it should use (avoiding the
+need for `--namespace`), disallow pushing to other peer's namespaces,
+and signing references.
+
+These three requirements can be solved by introducing a
+`git-remote-rad` helper binary that can supply the namespace, ensure
+the peer namespaces are respected, and that the signed references are
+updated. This is left to another RIP to define.
```
Comment (cloudhead) [inline 0002-storage-layout.md:324]: I'd actually say that we should move this little section up, as a solution to the above problems around verifying the push. We don't need to really specify the remote-helper that much; but I think this RIP is the place to do so. I'd talk about:
1. The `rad://` scheme
2. The handling of namespaces (ie. a user should not be able to push to a namespace other than his own)

**Review by cloudhead — COMMENTED** [review]
This is looking good, but I'm wondering if the scope is too narrow. There's two things I'd consider:

1. We should probably define now the formats of the user keys and project ids. I think this should probably go in RIP 1. This would allow this RIP to be much more specific about the storage layout, which I think will be really helpful to the reader.
2. I'd consider talking about signed-refs here, and more generally about `rad/*` special refs; because it is part of the storage layout. This might mean that this should be RIP 3, and RIP 2 should be abouy identity. I'm just trying to make this RIP more useful to implementors, and right now I think we're limiting ourselves too much due to unspecified stuff.

What do you think?

File: 0002-storage-layout.md [hunk 14]
```diff
@@ -0,0 +1,517 @@
...
+| macOS   | `$HOME/.radicle/storage`              | /Users/Alice/.radicle/storage   |
+| Windows | `{FOLDERID_Profile}\.radicle\storage` | C:\Users\Alice\.radicle\storage |
+
+Layout
+------
+
+The layout must support multiple resources and multiple peers per
+resource. Under the `storage` directory, each resource will be a [bare
+git repository][git-bare]. For each resource in the storage, it must
+have a stable and unique identifier. This is to ensure that resources
```
Comment (keepsimple1) [inline 0002-storage-layout.md:98]: Is this `unique identifier` globally unique, or only unique on this node?

File: 0002-storage-layout.md (hunk 8 above)
Comment (FintanH) [reply 0002-storage-layout.md:154]: But _should_ we say that here? That feels like it should be defined in a RIP that tackles replication directly, IMO.

File: 0002-storage-layout.md (hunk 11 above)
Comment (FintanH) [reply 0002-storage-layout.md:214]: I would argue that that's a decision made by the application, i.e. there isn't anything special about a `rad` remote, they could name it `local`, `origin`, or even `foobar`. But I can add a suggestion for doing so to make it easier to use.

File: 0002-storage-layout.md (hunk 14 above)
Comment (FintanH) [reply 0002-storage-layout.md:98]: Aye, it is. I'll clarify it in the doc.

File: 0002-storage-layout.md [hunk 15]
```diff
@@ -0,0 +1,517 @@
...
+interested in.
+
+Overview
+--------
+
+All nodes in the network must keep a local copy of the data they are
+interested in. For local users this improves the experience of
+interacting with the data -- not needing a server to compute data for
+you and being able to work offline. For local users and seeding nodes
+in the network this improves the availability of data.
```
Comment (slack-coder) [inline 0002-storage-layout.md:29]: Its also about censorship resistance

File: 0002-storage-layout.md (hunk 8 above)
Comment (cloudhead) [reply 0002-storage-layout.md:154]: Hmmm, think you're right.

File: 0002-storage-layout.md [hunk 16]
```diff
@@ -0,0 +1,517 @@
...
+
+Storage Home
+------------
+
+It is trivial to store `git` repositories, locally, on disk. It is
+also efficient to do so. The storage must be stored under the name
+`storage` found under one of two base directories.
+
+If the `RAD_HOME` environment variable is set, the directory it points
+to will be used as the base directory.
```
Comment (slack-coder) [inline 0002-storage-layout.md:76]: Maybe introducing what's being effected by RAD_HOME first will help readers?

Comment (slack-coder) [reply 0002-storage-layout.md:76]: for example, "RAD's storage is located by default under the user directory at X.   It can be overridden using the RAD_HOME envirnment variable."

It builds up to it using only defined concepts.

File: 0002-storage-layout.md [hunk 17]
```diff
@@ -0,0 +1,517 @@
+---
+RIP: 2
+Title: Storage Layout
+Author: '@fintohaps <fintan.halpenny@gmail.com>'
+Status: Draft
+Created: 2022-10-27
+License: CC0-1.0
+---
+
+RIP #2: Storage Layout
```
Comment (slack-coder) [inline 0002-storage-layout.md:10]: Would other parties making a 'RAD' be expected to use this storage format?

Comment (cloudhead) [reply 0002-storage-layout.md:10]: what's a RAD?

Comment (FintanH) [reply 0002-storage-layout.md:10]: If I'm understanding you correctly, the idea of a RIP is that these are expected implementations of any kind of Radicle clone. So if someone wrote a protocol and client in Go, they would be expected to do so following the RIPs to have parity with the Rust implementation.

File: 0002-storage-layout.md (hunk 16 above)
Comment (FintanH) [reply 0002-storage-layout.md:76]: > Maybe introducing what's being effected by RAD_HOME first will help readers?

The section is under "Storage Home", do you think that that is not clear enough? :thinking: 

I also finished the last paragraph with:
> The storage must be stored under the name `storage` found under one of two base directories.

But maybe I can say that is either the user's home or a preset `RAD_HOME`.

**Comment by FintanH** [comment]
Made a few changes in a fixup commit. Ready for another round of review :)

File: 0002-storage-layout.md [hunk 18]
```diff
@@ -204,20 +206,20 @@ A working copy is a local replica of the resource -- that is each
...
-wishes to use. For example, the `<pubkey>` of the peer could be used,
-or a nickname.
+The name, i.e. `[remote."name"]`, is open to be defined by the
+operator (or application), for example, the operator may use the
+public key of the peer, `origin`, `rad`, a nickname `finto`, etc.
 
-The `url` must be able to resolve the local storage's resource that
-this working copy corresponds to.
+The `url` of the remote must be able to resolve the local storage's
+resource that this working copy corresponds to.
```
Comment (cloudhead) [inline 0002-storage-layout.md:222]: "corresponding to this working copy" reads nicer I think

**Comment by FintanH** [comment]
Some more updates:
* Changed it to RIP#3 and rebased off of RIP#2 PR
* Edited links to mirror RIP#2 style
* Mentioned identifiers as established in RIP#2.
* Some wording fix ups.

File: 0003-storage-layout.md [hunk 19]
```diff
@@ -0,0 +1,564 @@
...
+```
+[remote "alice"]
+url = file:///path/to/storage
+fetch = +refs/remotes/<pubkey>/heads/*:refs/remotes/alice/*
+```
+
+This will do what you expect when running:
+
+```
+$ get fetch alice
```
Comment (gsaslis) [inline 0003-storage-layout.md:291]:
```suggestion
$ git fetch alice
```

... `get fetch` would be confusing in all sorts of weird ways, not to mention I'd never be able to remember if I `get fetch ...` or `fetch get ...`  ;p

File: 0003-storage-layout.md [hunk 20]
```diff
@@ -0,0 +1,564 @@
...
+```
+
+* The `rad://` scheme informs `git` to look for an executable
+  `git-remote-rad` which will be executed during a `git-push` or
+  `git-fetch`.
+* The `<resource>` component is the resource identifier to be found in
+  the storage.
+* The `<pubkey>` is the namespace for which the the `--namespace`
+  option will be set to. If `<pubkey>` is not specified, then the
+  local operator's key is used.
```
Comment (gsaslis) [inline 0003-storage-layout.md:348]: This whole section about how the `rad://` scheme works is very well explained, thanks for that!

File: 0003-storage-layout.md [hunk 21]
```diff
@@ -0,0 +1,564 @@
...
+
+The `url` scheme for a given remote is of the form:
+
+```
+rad://<resource>[/<pubkey>]
+```
+
+* The `rad://` scheme informs `git` to look for an executable
+  `git-remote-rad` which will be executed during a `git-push` or
+  `git-fetch`.
```
Comment (gsaslis) [inline 0003-storage-layout.md:343]: maybe a quick note could be added about where `git` would look for `git-remote-rad` ?

File: 0003-storage-layout.md [hunk 22]
```diff
@@ -0,0 +1,564 @@
...
+* The `<resource>` component is the resource identifier to be found in
+  the storage.
+* The `<pubkey>` is the namespace for which the the `--namespace`
+  option will be set to. If `<pubkey>` is not specified, then the
+  local operator's key is used.
+
+#### Authorization
+
+Since pushing to other peer's namespaces is disallowed, the value of
```
Comment (gsaslis) [inline 0003-storage-layout.md:352]: "disallowed" by whom? the Remote Helper? 

if so, perhaps we could reword this to sth like: 

> Since the Remote Helper MUST disallow pushing to other peer's namespaces ...

File: 0003-storage-layout.md [hunk 23]
```diff
@@ -0,0 +1,564 @@
...
+`<pubkey>` will be used to authorize whether a `git push` to a
+particular `url` is allowed. If the `<pubkey>` of the `url` does not
+match the local operator's key, then it must reject the push. Note
+that no such authorization is needed for a `git fetch`.
+
+#### Signing the Reference Set
+
+To ensure that the local operator's reference set is cryptographically
+verifiable, the reference set is signed with the operator's key during
```
Comment (gsaslis) [inline 0003-storage-layout.md:361]: it is a little unclear to me who does this signing ... I know we are in the remote helper section, but if it is the remote helper that takes care of this can we explicitly mention that here?

File: 0003-storage-layout.md [hunk 24]
```diff
@@ -0,0 +1,564 @@
...
+### Canonical References
+
+You may have noticed that in the new [layout](#Layout) the top-level
+namespace is left for canonical references. The definition and
+verification of canonicity is left for a future RIP.
+
+Appendix
+--------
+
+### Worked Example
```
Comment (gsaslis) [inline 0003-storage-layout.md:376]: Perhaps it is not within scope of this RIP, but it occurred to me that the example doesn't seem to cover signed commits. Considering there is earlier talk about signing happening during `git push` could we extend the example to cover that? (and if there is some provision to disallow pushing to other peer's namespaces it'd also be great if we showed how that happens!)

**Review by gsaslis — COMMENTED** [review]
@FintanH ok, so I don't know if this is the proper place to provide feedback, but I had some questions along the way as I was reading the doc to understand what's coming and I thought I'd drop some notes. 

It makes sense overall - I've just suggested some clarifications to be added and perhaps some wording improvements. 

Thanks for putting this together!

File: 0003-storage-layout.md [hunk 25]
```diff
@@ -0,0 +1,564 @@
...
+        * [Different Peers](#different-peers)
+        * [Non-global Tags](#non-global-tags)
+* [Credits](#credits)
+* [Copyright](#copyright)
+
+Storage Home
+------------
+
+It is trivial to store `git` repositories, locally, on disk. It is
+also efficient to do so. The storage must be stored under the name
```
[the remark this answers is not in the document]
Comment (FintanH) [reply 0003-storage-layout.md:74]: Aye, I believe that @cloudhead isn't a fan of that approach though :)

File: 0003-storage-layout.md [hunk 26]
```diff
@@ -0,0 +1,564 @@
...
+resource. Under the `storage` directory, each resource will be a bare
+git repository[^1]. For each resource in the storage, it must
+have a stable and globally unique identifier. This is to ensure that
+resources are stored uniquely in the `storage` and can be easily
+addressed by their identifier. This identifier is established as an
+RID in RIP#2's section on The Repository Identifier.
+
+```
+storage/
+  <RID>/   # Some resource, e.g. a project
```
[the remark this answers is not in the document]
Comment (FintanH) [reply 0003-storage-layout.md:109]: Please refer to [Zooko's triangle](https://en.wikipedia.org/wiki/Zooko%27s_triangle) :)

File: 0003-storage-layout.md (hunk 21 above)
Comment (FintanH) [reply 0003-storage-layout.md:343]: Ya, that's a good shout :+1:

File: 0003-storage-layout.md (hunk 22 above)
Comment (FintanH) [reply 0003-storage-layout.md:352]: Ya, essentially it's a protocol invariant. I'll make a note of this :)

File: 0003-storage-layout.md (hunk 24 above)
Comment (FintanH) [reply 0003-storage-layout.md:376]: Commit signing is out of scope. The protocol isn't concerned with signing individual commits, but if you do sign your commits then nothing here changes :)

File: 0002-storage-layout.md (hunk 17 above)
Comment (slack-coder) [reply 0002-storage-layout.md]: Thanks @FintanH.  Though the design may be good, would it be better to leave this out?  It looks to be an internal implementation detail.  Third party implementations would be considered a success right?  Minimizing RIPs to the protocol would reduce friction.

**Comment by gsaslis** [comment]
@FintanH another question: 

does Heartwood still (plan to) include "Collaborative Objects (Cobs)" ? If so, perhaps it would make sense to show how these are stored in this RIP ?

File: 0002-storage-layout.md (hunk 17 above)
Comment (FintanH) [reply 0002-storage-layout.md]: Why do you think the storage layout is an implementation detail?

**Comment by gsaslis** [comment]
> They are being used :) There's already existing code for them in Heartwood. That being said, I think that they should be defined in a separate RIP and how they are stored would follow the rules set out in this RIP, i.e. separate category under a specific peer, for each peer.

ok, ok, thanks! 👍🏼

File: 0002-storage-layout.md (hunk 17 above)
Comment (slack-coder) [reply 0002-storage-layout.md]: To be clear, its a question of Radicle Improvement Proposal's  over the design discussion.  Its good to have the design discussion for Heartwood in either case.

What are RIPs for?  They come across to me as design proposals for Radicle's Heartwood project.  Take defining `RAD_HOME` for example.  I recommend minimizing them to what's expected for interoperability for third party RAD applications.

RIPs appear to take from BIPs, and we missed an important aspect.  BIPs have three purposes as shown below.  Maybe we should adopt 'Process' as a use for RIPs?  The [first BIP](https://github.com/bitcoin/bips/blob/master/bip-0001.mediawiki) is about introducing BIPs, defining process and what they are for.

```
Standards Track BIPs – Such types of BIPs entail making changes to the [network](https://en.bitcoinwiki.org/wiki/Network) protocol, [block](https://en.bitcoinwiki.org/wiki/Block), or [transaction](https://en.bitcoinwiki.org/wiki/Transaction) validation method. It also intends to affect the interoperability of the two versions of BIPs or Bitcoin. This type of BIP certainly requires community consensus. An example of this is BIP 91.
    Informational BIPs – Such types of BIPs highlight the design issues, general guidelines, and supporting information. Informational BIPs, as the name suggests, are just for information’s sake and can be taken seriously or ignored by the community. An example of this is BIP 32.
    Process BIPs – These types of BIPs describe or propose a change in the process. They are similar to Standards Track BIPs and require community consensus. They can’t be ignored, but unlike Standards Track BIPs, they intend to be applied outside the [Bitcoin](https://en.bitcoinwiki.org/wiki/Bitcoin) protocol. An example of this is BIP 2.
```

Comment (slack-coder) [reply 0002-storage-layout.md]: > Why do you think the storage layout is an implementation detail?

The RAD_HOME environment variable, and the directory layout of its contents.  My perspective being what's needed for different users to interact with each other using Radicle, as opposed to different applications accessing Radicle project information for the same user.

Comment (slack-coder) [reply 0002-storage-layout.md]: > To be clear, its a question of Radicle Improvement Proposal's over the design discussion. Its good to have the design discussion for Heartwood in either case.
> 
> What are RIPs for? They come across to me as design proposals for Radicle's Heartwood project. Take defining `RAD_HOME` for example. I recommend minimizing them to what's expected for interoperability for third party RAD applications.
> 
> RIPs appear to take from BIPs, and we missed an important aspect. BIPs have three purposes as shown below. Maybe we should adopt 'Process' as a use for RIPs? The [first BIP](https://github.com/bitcoin/bips/blob/master/bip-0001.mediawiki) is about introducing BIPs, defining process and what they are for.
> 
>     * Standards Track BIPs – Such types of BIPs entail making changes to the [network](https://en.bitcoinwiki.org/wiki/Network) protocol, [block](https://en.bitcoinwiki.org/wiki/Block), or [transaction](https://en.bitcoinwiki.org/wiki/Transaction) validation method. It also intends to affect the interoperability of the two versions of BIPs or Bitcoin. This type of BIP certainly requires community consensus. An example of this is BIP 91.
> 
>     * Informational BIPs – Such types of BIPs highlight the design issues, general guidelines, and supporting information. Informational BIPs, as the name suggests, are just for information’s sake and can be taken seriously or ignored by the community. An example of this is BIP 32.
> 
>     * Process BIPs – These types of BIPs describe or propose a change in the process. They are similar to Standards Track BIPs and require community consensus. They can’t be ignored, but unlike Standards Track BIPs, they intend to be applied outside the [Bitcoin](https://en.bitcoinwiki.org/wiki/Bitcoin) protocol. An example of this is BIP 2.

To keep the topic to this RIP, l've created [an issue](https://github.com/radicle-dev/rips/issues/3) for fork this conversation to.

Comment (FintanH) [reply 0002-storage-layout.md]: I've removed mention of RAD_HOME and the storage home in general :) https://github.com/radicle-dev/rips/pull/1/commits/d301f209d688e45ee6d5da7fd1d6daab6b865ebf
